diff --git a/Foundation/Components/BaseMenu/BaseMenuItem.lua b/Foundation/Components/BaseMenu/BaseMenuItem.lua
index 82157a3..21d40ff 100644
--- a/Foundation/Components/BaseMenu/BaseMenuItem.lua
+++ b/Foundation/Components/BaseMenu/BaseMenuItem.lua
@@ -82,19 +82,22 @@ local function BaseMenuItem(menuItemProps: BaseMenuItemProps, ref: React.Ref<Gui
 		return BuilderIcons.Migration["uiblox"][props.icon]
 	end, { props.icon })
 
+	local viewProps = withCommonProps(props, {
+		GroupTransparency = if props.isDisabled then Constants.DISABLED_TRANSPARENCY else nil,
+		isDisabled = props.isDisabled,
+		onActivated = onActivated,
+		selection = {
+			Selectable = not props.isDisabled,
+		},
+		cursor = cursor,
+		tag = variantProps.container.tag,
+		ref = ref,
+	})
+	viewProps.ZIndex = 10
+
 	return React.createElement(
 		View,
-		withCommonProps(props, {
-			GroupTransparency = if props.isDisabled then Constants.DISABLED_TRANSPARENCY else nil,
-			isDisabled = props.isDisabled,
-			onActivated = onActivated,
-			selection = {
-				Selectable = not props.isDisabled,
-			},
-			cursor = cursor,
-			tag = variantProps.container.tag,
-			ref = ref,
-		}),
+		viewProps,
 		{
 			Icon = if props.icon or hasLeading
 				then if props.icon and isBuilderIconOrMigrated(props.icon)
@@ -118,6 +121,7 @@ local function BaseMenuItem(menuItemProps: BaseMenuItemProps, ref: React.Ref<Gui
 				LayoutOrder = 2,
 				Text = props.text,
 				tag = variantProps.text.tag,
+				ZIndex = 11,
 			}),
 			Check = if props.isChecked
 				then React.createElement(Icon, {
@@ -126,6 +130,7 @@ local function BaseMenuItem(menuItemProps: BaseMenuItemProps, ref: React.Ref<Gui
 					style = variantProps.check.style,
 					size = variantProps.check.size,
 					testId = `{props.testId}--checkmark`,
+					ZIndex = 11,
 				})
 				else nil,
 		}
